src/App.tsx

```diff
@@ -1,15 +1,51 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import { debounce } from 'lodash';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { Dropdown } from './components/Dropdown';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person>();
+  const [isVisible, setIsVisible] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filteredPeople, setIsFilteredPeople] = useState<Person[]>([]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const fetchFilteredPeople = useCallback(
+    debounce((inputValue: string) => {
+      const filterPeople = peopleFromServer.filter((person: Person) => (
+        person.name.toLowerCase().includes(inputValue.toLowerCase())
+      ));
+
+      setIsFilteredPeople(filterPeople);
+    }, 300), [],
+  );
+
+  const handleQueryChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const inputValue = event.target.value;
+
+      setQuery(inputValue);
+      setIsVisible(true);
+      setSelectedPerson(undefined);
+
+      fetchFilteredPeople(inputValue);
+    }, [fetchFilteredPeople],
+  );
+
+  const handlePersonSelected = useCallback((person: Person) => {
```

Comment:
no needs to wrap each function with `useCallback` only those that have a complex logic that can affect productivity
```suggestion
  const handlePersonSelected = (person: Person) => {
```